backend/lambdas/process_post/lambda.py

```diff
@@ -0,0 +1,193 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import os
+import traceback
+
+import demoji
+
+import boto3
+import langchain_core
+from langchain_aws import ChatBedrock
+
+from output_models.models import ExtractedInformation, TopicMatch
+from prompt_selector import get_information_extraction_prompt_selector, get_topic_match_prompt_selector
+
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from aws_lambda_powertools import Logger
+
+MODEL_ID = os.environ['MODEL_ID']
+AWS_REGION = os.environ['AWS_REGION']
+LANGUAGE_CODE = os.environ['LANGUAGE_CODE']
+META_TOPICS_STR = os.environ['LABELS']
+META_SENTIMENTS_STR = os.environ['SENTIMENT_LABELS']
+
+INFORMATION_EXTRACTION_PROMPT_SELECTOR = get_information_extraction_prompt_selector(LANGUAGE_CODE)
+TOPIC_MATCH_PROMPT_SELECTOR = get_topic_match_prompt_selector(LANGUAGE_CODE)
+
+TOPIC_MATCH_MODEL_PARAMETERS = {
+    "max_tokens": 500,
+    "temperature": 0.1,
+    "top_k": 20,
+}
+
+INFORMATION_EXTRACTION_MODEL_PARAMETERS = {
+    "max_tokens": 1500,
+    "temperature": 0.1,
+    "top_k": 20,
+}
+
+bedrock_runtime = boto3.client(
+    service_name="bedrock-runtime",
+    region_name=AWS_REGION
+)
+
+logger = Logger()
+
+langchain_core.globals.set_debug(True)
+
+def unknown_to_empty_text_values(extracted_info: ExtractedInformation):
+
+    extracted_info.topic = extracted_info.topic.strip()
+    extracted_info.location = extracted_info.location.strip()
+    extracted_info.sentiment = extracted_info.sentiment.strip()
+
+    extracted_info.topic = extracted_info.topic if extracted_info.topic != "<UNKNOWN>" else ""
+    extracted_info.location = extracted_info.location if extracted_info.location != "<UNKNOWN>" else ""
+    extracted_info.sentiment = extracted_info.sentiment if extracted_info.sentiment != "<UNKNOWN>" else ""
+
+
+def text_topic_match(
+        meta_topics: str,
+        text: str,
+) -> TopicMatch:
+
+    bedrock_llm = ChatBedrock(
+        model_id=MODEL_ID,
+        model_kwargs=TOPIC_MATCH_MODEL_PARAMETERS,
+        client=bedrock_runtime,
+    )
+
+    claude_topic_match_prompt_template = TOPIC_MATCH_PROMPT_SELECTOR.get_prompt(MODEL_ID)
+
+    print(claude_topic_match_prompt_template.format(meta_topics=meta_topics, text=text))
+
+    structured_llm = bedrock_llm.with_structured_output(TopicMatch)
+
+    structured_topic_match_chain = claude_topic_match_prompt_template | structured_llm
+
+    topic_match_obj = structured_topic_match_chain.invoke({"meta_topics": meta_topics, "text": text})
+
+    print("Topic match object")
+    print(type(topic_match_obj))
+    print(topic_match_obj)
+
+    return topic_match_obj
+
+def text_information_extraction(
+        sentiments: str,
+        text: str
+) -> ExtractedInformation:
+
+    bedrock_llm = ChatBedrock(
```

Comment:
Have you considered extracting this instantiation to root level so the client is created only once per lifetime of the lambda function?

Reply:
Cant extract from the function since the parameters are different for each instance of the client.